packages/perseus/src/widgets/cs-program/cs-program.tsx

```diff
@@ -108,11 +115,12 @@ class CSProgram extends React.Component<Props> {
         return Changeable.change.apply(this, args);
     };
 
+    getUserInput(): PerseusCSProgramUserInput {
+        return CSProgram.getUserInputFromProps(this.props);
+    }
```

Comment:
Ooo, for sure. I think I was just following the framework for other widgets that use `getUserInput`; it seems like most of them use this method. Maybe leave this for now and have a ticket to go through and refactor out `getUserInputFromProps` from as many widgets as possible? I'm not sure I understand why they were there in the first place.

Reply:
Okay, so first I kept the method and set up a test with it. I was following the example from categorizer's test. But then I tried to use the example from group's test and realized that cs-program uses an iframe. Checking out iframe, it doesn't have a `getUserInputFromProps` method, and it does it the way you suggest here. So I just did it this way in the end XD Not sure how to add a test though since iframe doesn't have any tests for `getUserInput` either. Do you have any suggestions @handeyeco ?

Reply:
Yeah, I don't really know. Maybe something like:

```TSX
// Arrange
const widgetRef: CSProgram | null = null;
render(<CSProgram ref={(ref) => widgetRef = ref} status="test status" message="test message />);

// Act
const result = widgetRef.getUserInput();

// Assert
expect(result.status).toBe("test status");
expect(result.message).toBe("test message");
```

I wouldn't lose sleep on it. We'll probably need to add tests to all the widgets for this and this could be a part of that process.